File: srcs/cookies/cookies.py
#!/usr/bin/python3

import os
import time
from http.cookies import SimpleCookie

# Timezone offset
timezone_offset = 3600

def parse_cookie(cookie_header = None):
    cookies = {}
    if not cookie_header:
        cookie_header = os.getenv('HTTP_COOKIE')
    cookies = SimpleCookie(cookie_header)
    return cookies

def set_cookie(cookie_name, cookie_value, expires, path = '/', domain = '', secure = False, httponly = True):

    set_cookie_header = f'Set-Cookie: {cookie_name}={cookie_value}; expires={time.strftime("%a, %d %b %Y %H:%M:%S GMT", time.gmtime(expires + timezone_offset))}; path=/'

    if domain:
        set_cookie_header += f"; domain={domain}"

    if secure:
        set_cookie_header += f"; secure"

    if httponly:
        set_cookie_header += f"; HttpOnly"

    print(f'{set_cookie_header}');

File: srcs/cookies/cookies.php
<?php

$timezone_offset = 3600;

function parse_cookie($cookie_header = null)
{
	if (!isset($cookie_header))
		$cookie_header = $_SERVER['HTTP_COOKIE'];
	$cookies = [];
	foreach (explode(';', $cookie_header) as $cookie) {
		list($name, $value) = explode('=', trim($cookie), 2) + [NULL, NULL];
		if ($name) {
			$cookies[$name] = $value;
		}
	}
	return $cookies;
}

function set_cookie($cookie_name, $cookie_value, $expires, $path = '/', $domain = '', $secure = false, $httponly = true)
{
	global $timezone_offset;

	$cookie_header = "Set-Cookie: $cookie_name=$cookie_value; expires=" . gmdate('D, d-M-Y H:i:s T', $expires + $timezone_offset) . "; path=$path";
	
	// Optionally add domain, secure, and httponly flags
	if ($domain) {
		$cookie_header .= "; domain=$domain";
	}
	if ($secure) {
		$cookie_header .= "; secure";
	}
	if ($httponly) {
		$cookie_header .= "; HttpOnly";
	}
	echo $cookie_header . PHP_EOL;
}
